lib/compiler.js

```diff
@@ -29,36 +79,97 @@ Compiler.prototype.generate = function generate(code, options) {
   options = options || {};
   code = fnToString(code);
 
-  var exportName = this.engineName.toUpperCase();
+  var exportName = options.exportName || this.engineName;
+
+  code = [
+    code + ';',
+    'oninit(function(exports, context) {',
+      'var BEMContext = exports.BEMContext || context.BEMContext;',
+      // Provides third-party libraries from different modular systems
+      'BEMContext.prototype.require = function(lib) {',
+        'return this._libs[lib];',
+      '};',
+    '});'
+  ].join('');
+
+  var deps = getDeps(options.requires);
 
   var source = [
-    '/// -------------------------------------',
-    '/// --------- BEM-XJST Runtime Start ----',
-    '/// -------------------------------------',
-    'var ' + exportName + ' = function(module, exports) {',
-      bundles[this.engineName],
-      ';',
-    '  return module.exports ||',
-    '      exports.' + exportName + ';',
-    '}({}, {});',
-    '/// -------------------------------------',
-    '/// --------- BEM-XJST Runtime End ------',
-    '/// -------------------------------------',
-    '',
-    'var api = new ' + exportName + '(' + JSON.stringify(options) + ');',
-    '/// -------------------------------------',
-    '/// ------ BEM-XJST User-code Start -----',
-    '/// -------------------------------------',
-    'api.compile(function(',
-    require('./bemxjst').prototype.locals.join(', '),
-    ') {',
-    getCode(code, options.runtimeLint) + ';',
-    '});',
-    'api.exportApply(exports);',
-    '/// -------------------------------------',
-    '/// ------ BEM-XJST User-code End -------',
-    '/// -------------------------------------\n'
-  ].join('\n');
+    'var ' + exportName + ';',
+    '(function(global) {',
+      'function buildBemXjst(libs) {',
+        'var exports;',
+
+        '/* BEM-XJST Runtime Start */',
```

Comment:
Do we really want replace these great beautiful banners with usual boring comment blocks?